metpy/calc/tools.py

```diff
@@ -157,3 +158,34 @@ def interpolate_nans(x, y, kind='linear'):
     else:
         raise ValueError('Unknown option for kind: {0}'.format(str(kind)))
     return y[x_sort_args]
+
+
+def _next_non_masked_element(a, idx):
+    """Return the next non masked element of a masked array.
+
+    If an array is masked, return the next non-masked element (if the given index is masked).
+    If no other unmasked points are after the given masked point, returns none.
+
+    Parameters
+    ----------
+    a : array-like
+        1-dimensional array of numeric values
+    idx : integer
+        index of requested element
+
+    Returns
+    -------
+        Index of next non-masked element and next non-masked element
+    """
+    if hasattr(a, 'mask'):
```

Comment:
Ever so slight gain in efficiency has convinced me of choosing EAFP over LBYL; code clarity, of course, dominates all of that, but that's not the case here.